src/main/java/at/hannibal2/skyhanni/features/garden/pests/PestFinder.kt

```diff
@@ -165,6 +165,18 @@ class PestFinder {
         if (lastKeyPress.passedSince() < 2.seconds) return
         lastKeyPress = SimpleTimeMark.now()
 
+        teleportNearestInfectedPlot()
+    }
+
+    @SubscribeEvent
+    fun onItemInHandChange(event: ItemInHandChangeEvent) {
+        if (!isEnabled()) return
+        if (!config.showPlotInWorld) return
+        if (event.oldItem !in PestAPI.vacuumVariants) return
+        lastTimeVacuumHold = SimpleTimeMark.now()
```

Comment:
this should go into pest api

Reply:
you mean the entire onKeyClick event? or just the teleportNearestInfectedPlot function

Reply:
the value lastTimeVacuumHold and the logic that calculates it

Reply:
ah, looked at the wrong thing.